### PR TITLE
Support an option to ignore pages from search index

### PR DESCRIPTION
**Pull request by ppwwyyxx**
Hi,

I'd like to ignore "pages" from search index and only search for "posts". This change is working for me.

Please let me know if this feature makes sense, and what else is needed to implement it. Thanks!

### CODE REVIEW

**Comment by ppoffice** [comment]
@ppwwyyxx Please fix the code format: https://github.com/ppoffice/hexo-component-inferno/runs/5995171676?check_suite_focus=true

File: package.json [hunk 1]
```diff
@@ -12,7 +12,7 @@
   "scripts": {
     "test": "jest lib",
     "test-cov": "jest lib --coverage",
-    "lint": "eslint {src,asset,fixture}/**/*.{js,jsx}",
+    "lint": "eslint '{src,asset,fixture}/**/*.{js,jsx}'",
```
Comment (ppwwyyxx) [inline package.json:15]: this is needed so lint can run inside `zsh`

**Comment by ppwwyyxx** [comment]
Thanks, I fixed the lint and `npm run lint; npm run test` passed locally.

**Comment by ppoffice** [comment]
Could you please also add unit tests for this new option?
https://github.com/ppoffice/hexo-component-inferno/blob/master/src/hexo/generator/insight.test.js

**Comment by ppwwyyxx** [comment]
Sure. A unittest is added.

**Pull request merged**